src/status_im/ui/screens/chat/components/input.cljs

```diff
@@ -79,6 +79,81 @@
+  (let [selection (.-selection ^js (.-nativeEvent ^js args))
+        start     (.-start selection)
+        end       (.-end selection)]
+    ;; NOTE(rasom): on iOS we do not dispatch this event immediately
+    ;; because it is needed only in case if selection is changed without
+    ;; typing. Timeout might be canceled on `on-change`.
+    (when platform/ios?
+      (reset!
+       timeout-id
+       (utils.utils/set-timeout
+        #(re-frame/dispatch [::mentions/on-selection-change
+                             {:start start
+                              :end   end}
+                             mentionable-users])
+        50)))
+    ;; NOTE(rasom): on Android we dispatch event only in case if there
+    ;; was no text changes during last 50ms. `on-selection-change` is
+    ;; dispatched after `on-change`, that's why there is no another way
+    ;; to know whether selection was changed without typing.
+    (when (and platform/android?
+               (or (not @last-text-change)
+                   (< 50 (- (js/Date.now) @last-text-change))))
+      (re-frame/dispatch [::mentions/on-selection-change
+                          {:start start
+                           :end   end}
+                          mentionable-users]))
+    ;; NOTE(rasom): we have to reset `cursor` value when user starts using
+    ;; text-input because otherwise cursor will stay in the same position
+    (when (and cursor platform/android?)
+      (re-frame/dispatch [::mentions/clear-cursor]))))
+
+(defn on-change [on-text-change last-text-change timeout-id  mentionable-users args]
+  (let [text (.-text ^js (.-nativeEvent ^js args))]
+    ;; NOTE(rasom): on iOS `on-selection-change` is canceled in case if it
+    ;; happens during typing because it is not needed for mention
+    ;; suggestions calculation
+    (when (and platform/ios? @timeout-id)
+      (utils.utils/clear-timeout @timeout-id))
+    (when platform/android?
+      (reset! last-text-change (js/Date.now)))
+    (on-text-change text)
+    ;; NOTE(rasom): on iOS `on-change` is dispatched after `on-text-input`,
+    ;; that's why mention/channel suggestions are calculated on `on-change`
+    (when platform/ios?
+      (re-frame/dispatch [::mentions/calculate-suggestions mentionable-users]))))
+
+(defn on-text-input [mentionable-users args]
+  (let [native-event  (.-nativeEvent ^js args)
+        text          (.-text ^js native-event)
+        previous-text (.-previousText ^js native-event)
+        range         (.-range ^js native-event)
+        start         (.-start ^js range)
+        end           (.-end ^js range)]
+    (re-frame/dispatch
+     [::mentions/on-text-input
+      {:new-text      text
+       :previous-text previous-text
+       :start         start
+       :end           end}])
+    ;; NOTE(rasom): on Android `on-text-input` is dispatched after
+    ;; `on-change`, that's why mention/channel suggestions are calculated
```

Comment:
we can just say "suggestions" instead of "mention/channel suggestions" here and in the rest of comments